content/blog/automated-releases.md

```diff
@@ -0,0 +1,346 @@
+
+The first workflow reacts to changes in the release branch (`master` in this case) and decides if release should be triggered, and triggers it. The last step of the workflow is a pull request creation with changes in `package.json`. Why are changes not committed directly to the release branch? Because in the majority of cases, people, like we in AsyncAPI, use branch protection rules and do not allow direct commits to release branches.
+
+The second workflow is just for handling changes in `package.json`. To fulfill branch protection settings, we had to auto-approve the pull request so we can automatically merge it.
+
+You can extend this workflow with additional steps, like:
+
+* Integration testing
+* Deployment
+* Notifications
+
+## GitHub Actions
+
+Even though I have my opinion about GitHub Actions (for more details, you can read my [post about it](https://dev.to/derberg/github-actions-when-fascination-turns-into-disappointment-4d75)) I still think it is worth investing in it, especially for the release workflows.
+
+We used the GitHub-provided actions and the following awesome actions provided by the community:
+
+- [Create Pull Request](ttps://github.com/marketplace/actions/create-pull-request)
+- [Auto Approve](https://github.com/marketplace/actions/auto-approve)
+- [Merge Pull Request](https://github.com/marketplace/actions/merge-pull-requests)
+
+### Release workflow
+
+Release workflow triggers every time there is something new happening in the release branch. In our case, it is a `master` branch:
+
+```yaml
+on:
+  push:
+    branches:
+      - master
+```
+
+#### GitHub and NPM
+
+For releases to GitHub and NPM, the most convenient solution is to integrate [semantic release](https://github.com/semantic-release/semantic-release) package and related plugins that support Conventional Commits. You can configure plugins in your `package.json` in the order they should be invoked:
+
+```json
+"plugins": [
+  [
+    "@semantic-release/commit-analyzer",
+    {
+      "preset": "conventionalcommits"
+    }
+  ],
+  [
+    "@semantic-release/release-notes-generator",
+    {
+      "preset": "conventionalcommits"
+    }
+  ],
+  "@semantic-release/npm",
+  "@semantic-release/github"
+]
+```
+
+Remember that functional automation should use a technical bot rather than a real user. GitHub Actions allow you to encrypt credentials to different systems on the repository level and referring them in actions looks as follows: 
+
+```yaml
+- name: Release to NPM and GitHub
+  id: release
+  env:
+    GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+    NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+    GIT_AUTHOR_NAME: asyncapi-bot
+    GIT_AUTHOR_EMAIL: info@asyncapi.io
+    GIT_COMMITTER_NAME: asyncapi-bot
+    GIT_COMMITTER_EMAIL: info@asyncapi.io
+  run: npm run release
+```
+
+The great bonus thing you get from this toolset is that at the end, once your pull request is merged and released, you get helpful information under it that it is a part of the release:
```

Comment:
```suggestion
Aside from automation, the bot will also comment on every pull request included in the release notifying participants their code has been shipped. Isn't it awesome?
```

My suggestion to rephrase this sentence.

Reply:
I will change to below to eliminate will and to also include issues. what you think?

```
Aside from automation, the bot also comment on every pull request and issue included in the release notifying subscribed participants that the given topic is part of the release. Isn't it awesome?
```

Reply:
Yup. Just notice it should be "the bot also **comments**".